src/main/java/mate/academy/MergeSortAction.java

```diff
@@ -1,8 +1,74 @@
 package mate.academy;
 
-public class MergeSortAction {
+import java.util.concurrent.RecursiveAction;
+
+public class MergeSortAction extends RecursiveAction {
+    private static final int THRESHOLD = 2;
+    private final int[] array;
+    private final int left;
+    private final int right;
 
     public MergeSortAction(int[] array) {
+        this(array, 0, array.length - 1);
+    }
+
+    public MergeSortAction(int[] array, int left, int right) {
+        this.array = array;
+        this.left = left;
+        this.right = right;
+    }
+
+    @Override
+    protected void compute() {
+        if (right - left + 1 <= THRESHOLD) {
```

Comment:
The threshold for switching to insertion sort is set to 2. This is quite low and may not provide the expected performance benefits of using insertion sort for small subarrays. Consider increasing this threshold to a higher value, such as 10 or 20, to better utilize insertion sort's efficiency on small datasets.